From b152b3e0c142cffab703d9247da5fe5d3698947a Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Tue, 24 Sep 2019 19:43:23 +0530
Subject: [PATCH 0291/1921] octeontx2-af: Init mcam resources after loading
 profile

Initialize mcam resources like key width after a
profile is loaded because loaded profile may
change key width.

Change-Id: Ia372ab9357efe6e1cb879ba3e6b8b5764b2eeb88
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/16397
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c     | 2 +-
 drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 41a088bdabfb..534c64c8ddff 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -739,7 +739,7 @@ static int rvu_fwdata_init(struct rvu *rvu)
 	}
 	return 0;
 fail:
-	dev_err(rvu->dev, "Unable to fetch 'fwdata' from firmware\n");
+	dev_info(rvu->dev, "Unable to fetch 'fwdata' from firmware\n");
 	return -EIO;
 }
 
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
index 60c15dafbd54..47874c7bb87e 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
@@ -1480,13 +1480,13 @@ int rvu_npc_init(struct rvu *rvu)
 	rvu_write64(rvu, blkaddr, NPC_AF_INTFX_KEX_CFG(NIX_INTF_TX),
 			((keyz & 0x3) << 32) | nibble_ena);
 
+	/* Configure MKEX profile */
+	npc_load_mkex_profile(rvu, blkaddr);
+
 	err = npc_mcam_rsrcs_init(rvu, blkaddr);
 	if (err)
 		return err;
 
-	/* Configure MKEX profile */
-	npc_load_mkex_profile(rvu, blkaddr);
-
 	err = npc_flow_steering_init(rvu, blkaddr);
 	if (err) {
 		dev_err(rvu->dev,
-- 
2.31.1

